### ISSUE TITLE
As bigger blockchain gets, longer restart takes - Rebuilding blockchain become inefficient

### ISSUE THREAD

**Issue by karek314** [LiskHQ/lisk#232]
As bigger blockchain gets, longer restart takes - Rebuilding blockchain become inefficient and takes more and more time to finish.

**Comment by Gr33nDrag0n69** [comment]
The solution to this issue is simple and complicated. We need official snapshots to be able to rebuild faster but issue were found about the snapshots creations in v0.2.x. So, for me,  the real issue is how to build safe official snapshots and release them at least once a week or 2. (if the snapshots must be verified and the job take some time to do (short term).

Another solution that could be implemented short term is the block/unblock of port 7000/8000 incoming connections automatically when rebuild is in progress. The unblock command could be based on the latest received block age. (like < 5 min the port is unblocked)

I think they are on the right path because obviously working on stability will minimize the number of rebuild at first ...

**Comment by karek314** [comment]
I have noticed that "Rebuilding blockchain" happens each time node is restarted, why is that? Why node do not trust previously obtained blockchain? I haven't checked that part of code, what is purpose?

I see you bring here "official snapshots" topic here, this is thing which concerns me, i think it would be much better to improve network sync instead of introducing thing called "official snapshots". Another solution which will be decentralised is "decentralised snapshots" each node which is willing to help network sync at the maximum level could be simply adding one flag in config which allow to create and distribute "decentralised snapshots" as small packages of blockchain - for example zipped parts of blockchain. Each 5000 blocks new package is being created and newly introduced node could simply at efficient speed acquire all packages, uncompress and start rebuilding and verification while another one is downloading from this or another node. But similar solution could be achieved with improvements to network sync, since sometimes node is syncing very fast while sometimes it slows down to one block per minute which is slower than network creates blocks, then after few hours it become super-fast again but not for long which forces me to restart nodes few times to keep fast speed of sync.

**Comment by fix** [comment]
This is a common issue for all crypto.

Snapshot while usefull for most of the users is NOT recommended for delegates.

There is also 2 issues:
- time to process block
- getting slow when the rebuilding go on.

The latter is not fully identified but is fixed when closing the ports 7000 or 8000
So fqr, it seems that there are a lot of transactions getting in an stacked into the node. On each block processed, thoses tx are poped from the database and pushed again at the end, making the whole process really slow and unefficient.

Closing as duplicate of #147

**Comment by karek314** [comment]
> This is a common issue for all crypto.
Im aware of this, but in this case it's becoming incredibly to slow with time.

According to described issue i don't see problem with blocking client functionality while blockchain is not synced by displaying Blockchain loading message with progress bar, maybe speed and some more informations, maybe also some guides or thing to read/learn.
